src/test/java/it/gov/pagopa/idpay/transactions/event/CommandsConsumerConfigIntegrationTest.java

```diff
@@ -0,0 +1,190 @@
+        long timePublishingEnd = System.currentTimeMillis();
+
+        waitForLastStorageChange((validMessages/VALID_USE_CASES)*2*NUMBER_TRX);
+
+        long timeEnd=System.currentTimeMillis();
+
+        checkRepositories();
+        checkErrorsPublished(notValidMessages, maxWaitingMs, errorUseCases);
+
+        System.out.printf("""
+                        ************************
+                        Time spent to send %d (%d + %d) messages (from start): %d millis
+                        Time spent to assert db stored count (from previous check): %d millis
+                        ************************
+                        Test Completed in %d millis
+                        ************************
+                        """,
+                commandsPayloads.size(),
+                validMessages,
+                notValidMessages,
+                timePublishingEnd - timeStart,
+                timeEnd - timePublishingEnd,
+                timeEnd - timeStart
+        );
+
+        long timeCommitCheckStart = System.currentTimeMillis();
+        Map<TopicPartition, OffsetAndMetadata> srcCommitOffsets = kafkaTestUtilitiesService.checkCommittedOffsets(topicCommands, groupIdCommands, commandsPayloads.size());
+        long timeCommitCheckEnd = System.currentTimeMillis();
+
+        System.out.printf("""
+                        ************************
+                        Time occurred to check committed offset: %d millis
+                        ************************
+                        Source Topic Committed Offsets: %s
+                        ************************
+                        """,
+                timeCommitCheckEnd - timeCommitCheckStart,
+                srcCommitOffsets
+        );
+
+    }
+
+
+    private long waitForLastStorageChange(int n) {
+        long[] countSaved={0};
+        //noinspection ConstantConditions
+        TestUtils.waitFor(()->(countSaved[0]=rewardTransactionRepository.findAll().count().block()) == n, ()->"Expected %d saved transactions in db, read %d".formatted(n, countSaved[0]), 60, 1000);
+        return countSaved[0];
+    }
+
+    private List<String> buildValidPayloads(int startValue, int messagesNumber) {
+        return IntStream.range(startValue, messagesNumber)
+                .mapToObj(i -> {
+                    QueueCommandOperationDTO command = QueueCommandOperationDTO.builder()
+                            .entityId(INITIATIVEID.formatted(i))
+                            .operationTime(LocalDateTime.now())
+                            .build();
+                    switch (i%VALID_USE_CASES){
+                        case 0 -> {
+                            INITIATIVES_DELETED.add(command.getEntityId());
+                            command.setOperationType(CommandsConstants.COMMANDS_OPERATION_TYPE_DELETE_INITIATIVE);
+                            initializeDB(i, "QRCODE");
```

Comment:
is command.getEntityId the initiativeId? why are you not passing it to initializeDb method instead of calculate again the initiativeId?